webodf/lib/gui/ZoomHelper.js

```diff
@@ -443,13 +443,31 @@
         };
 
         /**
-         * @param {!number} zoomLevel
-         * @return {undefined}
-         */
+        * Sets the zoom level of the document and maintains the top left position
+        * @param {!number} zoomLevel
+        * @return {undefined}
+        */
         this.setZoomLevel = function (zoomLevel) {
             if (zoomableElement) {
+                var scrollRatios = [];
+                var currentOffsetParent = offsetParent;
+                // Save the scroll ratios and related parent
+                while (currentOffsetParent) {
+                    scrollRatios.push({ container: currentOffsetParent, scrollTop: currentOffsetParent.scrollTop / zoom,
+                                        scrollLeft: currentOffsetParent.scrollLeft / zoom });
+                    currentOffsetParent = currentOffsetParent.offsetParent;
+                }
+
                 zoom = zoomLevel;
                 applyDetailedTransform();
+
+                // Restore the scroll ratios
+                currentOffsetParent = offsetParent;
+                scrollRatios.forEach(function(savedRatio) {
+                    currentOffsetParent.scrollTop = savedRatio.scrollTop * zoom;
+                    currentOffsetParent.scrollLeft = savedRatio.scrollLeft * zoom;
+                    currentOffsetParent = currentOffsetParent.offsetParent;
```

Comment:
This variable shouldn't be required now, as you should be using `savedRatio.container`